.github/workflows/check-version.yml

```diff
@@ -0,0 +1,87 @@
+name: Check version
+
+on:
+  push:
+    branches:
+      - develop
+      - release/*
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  VERSION_FILE: ./package.json
+
+jobs:
+  check-version:
+    name: Check version
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Get current version
+        id: current_version
+        run: |
+          currentVersion=$(node -p -e "require('${{ env.VERSION_FILE }}').version")
+
+          echo "version=$currentVersion" >> $GITHUB_OUTPUT
+
+      - name: Get latest version
+        id: latest_version
+        uses: pozetroninc/github-action-get-latest-release@v0.7.0
+        with:
+          repository: nimblehq/infrastructure-templates
+
+      - name: Compare versions
+        id: compare_versions
+        run: |
+          if [[ "${{ steps.latest_version.outputs.release }}" != "${{ steps.current_version.outputs.version }}" ]]; then
+            exit 0
+          else
+            echo "The latest version is the same as the current version."
+            exit 1
+          fi
```

Comment:
Bumping a new version on `main` can help us to ensure only running to bump version after merging the release to the main to finish the screen. Hence, this compare version step could be eliminated 💡

Reply:
Changed in https://github.com/nimblehq/infrastructure-templates/pull/237/commits/3b0ed365bfb509d6ebd0f30f4d99a58704f62951